src/main/java/at/hannibal2/skyhanni/features/garden/SensitivityReducer.kt

```diff
@@ -3,218 +3,133 @@ package at.hannibal2.skyhanni.features.garden
-    private const val LOCKED = -1F / 3F
 
-    private val gameSettings get() = Minecraft.getMinecraft().gameSettings
-    private val playerOnGround get() = MinecraftCompat.localPlayer.onGround
+    private var inBarn: Boolean = false
+    private var onGround: Boolean = false
+
+    private var shouldBeActive = false
+    private val isActive get() = isAutoActive || isManualActive
+    private val isAutoActive get() = MouseSensitivityHook.state == MouseSensitivityHook.MouseSensitivityState.AUTO_REDUCED
+    private val isManualActive get() = MouseSensitivityHook.state == MouseSensitivityHook.MouseSensitivityState.MANUAL_REDUCED
+    private val isMouseLocked get() = MouseSensitivityHook.state == MouseSensitivityHook.MouseSensitivityState.LOCKED
 
-    @Suppress("CyclomaticComplexMethod")
     @HandleEvent
     fun onTick() {
         if (!GardenApi.inGarden()) {
-            if (isToggled && lastCheckCooldown.passedSince() > 1.seconds) {
-                lastCheckCooldown = SimpleTimeMark.now()
-                isToggled = false
-                restoreSensitivity()
-            }
-            return
-        }
-        if (isManualToggle) return
-        if (isToggled && config.onGround.get() && !playerOnGround) {
-            restoreSensitivity()
-            isToggled = false
+            if (isAutoActive) autoToggle()
             return
         }
-        when (config.mode) {
-            SensitivityReducerConfig.Mode.OFF -> {
-                if (isToggled) toggle(false)
-                return
-            }
-
-            SensitivityReducerConfig.Mode.TOOL -> {
-                if (isHoldingTool() && !isToggled) toggle(true)
-                else if (isToggled && !isHoldingTool()) toggle(false)
-            }
-
-            SensitivityReducerConfig.Mode.KEYBIND -> {
-                if (isHoldingKey() && !isToggled) toggle(true)
-                else if (isToggled && !isHoldingKey()) toggle(false)
-            }
-
-            else -> return
-        }
-        if (isToggled && lastCheckCooldown.passedSince() > 1.seconds) {
-            if (GardenApi.onBarnPlot && config.onlyPlot.get()) {
-                isToggled = false
-                restoreSensitivity()
-            }
-            if (!playerOnGround && config.onGround.get()) {
-                isToggled = false
-                restoreSensitivity()
-            }
-            lastCheckCooldown = SimpleTimeMark.now()
-        }
+        if (isMouseLocked) return
+
+        updatePlayerStatus()
+        autoToggleIfNeeded()
     }
 
-    @HandleEvent
-    fun onConfigLoad(event: ConfigLoadEvent) {
-        config.reducingFactor.afterChange {
-            reloadSensitivity()
-        }
-        config.onlyPlot.afterChange {
-            if (isToggled && config.onlyPlot.get() && GardenApi.onBarnPlot) {
-                restoreSensitivity()
-                isToggled = false
-            }
+    private fun updatePlayerStatus() {
```

Comment:
compact this function by comparing `true != false` instead of `true && !true` and `false && !false`